src/coreclr/jit/compiler.cpp

```diff
@@ -5553,7 +5553,9 @@ int Compiler::compCompile(CORINFO_MODULE_HANDLE classPtr,
     compFrameInfo = {0};
 #endif
 
-    virtualStubParamInfo = new (this, CMK_Unknown) VirtualStubParamInfo(IsTargetAbi(CORINFO_NATIVEAOT_ABI));
+    bool isNativeAot     = IsTargetAbi(CORINFO_NATIVEAOT_ABI);
+    virtualStubParamInfo = new (this, CMK_Unknown) VirtualStubParamInfo(isNativeAot);
+    m_preferredInitCctor = CORINFO_HELP_READYTORUN_NONGCSTATIC_BASE;
```

Comment:
Now that this doesn't need to check for NativeAOT ABI it would be better to revert this change and initialize it in `compInit`